### DISCUSSION TITLE
How can I check if the C extensions are installed on a system?

### DISCUSSION THREAD

**Discussion by LostInDarkMath** [sqlalchemy/sqlalchemy#6278] [Usage Questions] [answered]
I wonder if there is a way to figure out if the C extensions are installed. How can I do that
- on Windows?
- on Linux?

**Comment by CaselIT** [comment, marked as answer]
Hi,

If you install from pip on a compatible architecture the wheel have c extensions active.

That said you can look for 1.4 at https://github.com/sqlalchemy/sqlalchemy/blob/ea225c63f98dc3c250cee1b243762970b8e0932a/.github/workflows/create-wheels.yaml#L90 and for 1.3 at https://github.com/sqlalchemy/sqlalchemy/blob/7dd9200c01644f735c9a3a76337e715bae350667/.github/workflows/create-wheels.yaml#L91 

note that these api are to be considered private, so they may change in the future.

**Reply by LostInDarkMath** [reply]
Thanks for your fast reply! But I'm not sure if this answers my question. Did you mean that C extensions are always installed, when I install SQLAlchemy, even if there is no C compiler installed?

**Reply by CaselIT** [reply]
yes, if you install from pypi and pip installs the wheel package, the c-extension are installed, since the wheel are pre-compiled.

if you install with `pip install --no-binary` (of the architecture is not supported, like alpine)  then they are compiled if there is a compiler or skipped if there is none
